File: clear_users.py
#!/usr/bin/env python3
"""
Script to clear users from the database for testing.
"""
import os
from pathlib import Path

# Load environment variables from .env.local file
def load_env():
    env_file = Path(__file__).parent / '.env.local'
    if env_file.exists():
        with open(env_file) as f:
            for line in f:
                line = line.strip()
                if line and not line.startswith('#') and '=' in line:
                    key, value = line.split('=', 1)
                    # Remove quotes from values
                    value = value.strip('"')
                    os.environ[key] = value

load_env()

try:
    import psycopg2
    from psycopg2 import sql
except ImportError:
    print("psycopg2 not installed. Installing...")
    os.system("pip install psycopg2-binary")
    import psycopg2
    from psycopg2 import sql

def clear_users():
    """Clear all users from the database."""
    db_url = os.getenv('DATABASE_URL')
    if not db_url:
        print("❌ DATABASE_URL not found in environment")
        return False
    
    try:
        # Parse the database URL
        import urllib.parse
        parsed = urllib.parse.urlparse(db_url)
        
        print(f"Connecting to database: {parsed.hostname}:{parsed.port}/{parsed.path[1:]}")
        
        # Connect to database
        conn = psycopg2.connect(
            host=parsed.hostname,
            port=parsed.port,
            database=parsed.path[1:],
            user=parsed.username,
            password=parsed.password
        )
        
        cursor = conn.cursor()
        
        # Check current users
        cursor.execute("SELECT COUNT(*) FROM users;")
        user_count = cursor.fetchone()[0]
        print(f"📊 Current users in database: {user_count}")
        
        if user_count > 0:
            # Show existing users
            cursor.execute("SELECT id, name, email, email_verified, created_at FROM users ORDER BY created_at;")
            users = cursor.fetchall()
            
            print("\n📋 Current users:")
            for user in users:
                print(f"  - ID: {user[0]}, Name: {user[1]}, Email: {user[2]}, Verified: {user[3]}, Created: {user[4]}")
            
            # Clear users
            print(f"\n🗑️  Removing {user_count} users...")
            cursor.execute("DELETE FROM users;")
            conn.commit()
            
            print("✅ All users removed successfully!")
        else:
            print("✅ Database is already empty (no users found)")
        
        cursor.close()
        conn.close()
        return True
        
    except Exception as e:
        print(f"❌ Database error: {e}")
        return False

if __name__ == "__main__":
    print("🧹 Clearing users from database...")
    success = clear_users()
    if success:
        print("\n✅ Database cleared. Ready for testing signup workflow!")
    else:
        print("\n❌ Failed to clear database. Please check database connection.")

File: clear_users.js
#!/usr/bin/env node
/**
 * Script to clear users from Supabase database for testing.
 */

const fs = require('fs');
const { createClient } = require('@supabase/supabase-js');

// Load environment variables manually from .env.local
function loadEnv() {
  try {
    const envContent = fs.readFileSync('.env.local', 'utf8');
    envContent.split('\n').forEach(line => {
      line = line.trim();
      if (line && !line.startsWith('#') && line.includes('=')) {
        const [key, value] = line.split('=', 2);
        process.env[key] = value.replace(/^"|"$/g, ''); // Remove quotes
      }
    });
  } catch (error) {
    console.log('Warning: Could not load .env.local file');
  }
}

loadEnv();

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.log('❌ Missing Supabase credentials in .env.local');
  console.log('NEXT_PUBLIC_SUPABASE_URL:', !!supabaseUrl);
  console.log('SUPABASE_SERVICE_ROLE_KEY:', !!supabaseKey);
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function clearUsers() {
  console.log('🧹 Checking users from Supabase...');
  
  try {
    // First, check if we can connect and see current users
    const { data: users, error: fetchError } = await supabase
      .from('users')
      .select('*')
      .order('created_at');
    
    if (fetchError) {
      console.log('❌ Error fetching users:', fetchError.message);
      return false;
    }
    
    console.log(`📊 Current users in database: ${users.length}`);
    
    if (users.length > 0) {
      console.log('\n📋 Current users:');
      users.forEach(user => {
        console.log(`  - ID: ${user.id}`);
        console.log(`    Name: ${user.name || 'N/A'}`);
        console.log(`    Email: ${user.email}`);
        console.log(`    Verified: ${user.email_verified || false}`);
        console.log(`    Verification Token: ${user.verification_token || 'N/A'}`);
        console.log(`    Token Expires: ${user.verification_expires || 'N/A'}`);
        console.log(`    Created: ${user.created_at}`);
        console.log('');
      });
      
      // Clear all users
      console.log(`🗑️  Removing ${users.length} users...`);
      const { error: deleteError } = await supabase
        .from('users')
        .delete()
        .neq('id', '00000000-0000-0000-0000-000000000000'); // Delete all users
      
      if (deleteError) {
        console.log('❌ Error deleting users:', deleteError.message);
        return false;
      }
      
      console.log('✅ All users removed successfully!');
    } else {
      console.log('✅ Database is already empty (no users found)');
    }
    
    return true;
    
  } catch (error) {
    console.log('❌ Unexpected error:', error.message);
    return false;
  }
}

async function main() {
  console.log('🔗 Connecting to Supabase...');
  console.log('URL:', supabaseUrl);
  console.log('Key:', supabaseKey ? 'Present' : 'Missing');
  
  const success = await clearUsers();
  
  if (success) {
    console.log('\n✅ Database cleared. Ready for testing signup workflow!');
    console.log('\n📝 Next steps:');
    console.log('   1. Start the development server: npm run dev');
    console.log('   2. Navigate to: http://localhost:3000/auth/signup');
    console.log('   3. Test the complete signup flow');
  } else {
    console.log('\n❌ Failed to clear database. Please check Supabase connection.');
  }
}

if (require.main === module) {
  main().catch(console.error);
}